util/util.go

```diff
@@ -0,0 +1,63 @@
+package util
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var fullCommandRegexp = regexp.MustCompile(`^([a-zA-Z0-9-_]+)\/([a-zA-Z0-9-_]+)@([a-z0-9-~\*\^\.]+)$`)
+var xrangesRegexp = regexp.MustCompile(`^(?:(\d+)\.)?(?:(\d+)\.)?([\*x]|\d+)$`)
+var tildeRangesRegexp = regexp.MustCompile(`^~\d(\.\d)?(\.\d)?$`)
+var caretRangesAndPinningRegexp = regexp.MustCompile(`^(\^)?\d(\.\d){2}$`)
+var tagRegexp = regexp.MustCompile(`^[a-z0-9-]+$`)
+
+func checkVersion(ver string) bool {
+	if caretRangesAndPinningRegexp.Match([]byte(ver)) {
+		return true
+	}
+	if tildeRangesRegexp.Match([]byte(ver)) {
+		return true
+	}
+	if xrangesRegexp.Match([]byte(ver)) {
+		return true
+	}
+	if tagRegexp.Match([]byte(ver)) {
+		return true
+	}
+	return false
+}
+
+// SplitCmd splits full command to namespace, command, version.
+// ex(ns/cmd/1.0.1 => ns cmd 1.0.1)
+func SplitCmd(cmd string) (namespace, command, version string, err error) {
+	values := fullCommandRegexp.FindAllStringSubmatch(cmd, -1)
+	if len(values) < 1 {
+		err = fmt.Errorf("There is no full command")
+		return
+	}
+
+	if len(values[0]) != 4 {
+		err = fmt.Errorf("The full command has somthing wrong")
```

Comment:
something
If you want to use "something wrong", you should use like: "There is something wrong with the full command"

Reply:
Thank you. I fixed